cardano-db-sync/src/Cardano/DbSync/Era/Universal/Insert/Other.hs

```diff
@@ -181,37 +206,41 @@ insertMultiAsset cache policy aName = do
 
 insertScript ::
   (MonadBaseControl IO m, MonadIO m) =>
-  Trace IO Text ->
+  SyncEnv ->
   DB.TxId ->
   Generic.TxScript ->
-  ReaderT SqlBackend m DB.ScriptId
-insertScript tracer txId script = do
-  mScriptId <- DB.queryScript $ Generic.txScriptHash script
-  case mScriptId of
-    Just scriptId -> pure scriptId
-    Nothing -> do
-      json <- scriptConvert script
-      DB.insertScript $
-        DB.Script
-          { DB.scriptTxId = txId
-          , DB.scriptHash = Generic.txScriptHash script
-          , DB.scriptType = Generic.txScriptType script
-          , DB.scriptSerialisedSize = Generic.txScriptPlutusSize script
-          , DB.scriptJson = json
-          , DB.scriptBytes = Generic.txScriptCBOR script
-          }
+  ReaderT SqlBackend m (Maybe DB.ScriptId)
+insertScript syncEnv txId script =
+  if isSimplePlutusScriptHashInWhitelist syncEnv $ Generic.txScriptHash script
```

Comment:
I think this check should change. See the comment at `isSimplePlutusScriptHashInWhitelist` for more.